### PR TITLE
examples(with-mdx): update to MDX 3

### PR DESCRIPTION
**Pull request by remcohaszing**
### What?

Update the `with-mdx` example to MDX 3.

### Why?

The example was 2 major versions behind.

### How?

By updating the version in `package.json`. Also the `@mdx-js/react` dependency was removed, as it doesn’t work with the app directory. Users should use `mdx-components.tsx` instead.

Refs #59693

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: e4079f94527744a76a86beca87e1c66a0266db18

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/with-mdx/package.json
Comment (leerob) [inline examples/with-mdx/package.json:10]: There's a separate example for app router, should this dep remain?

https://github.com/vercel/next.js/tree/canary/examples/app-dir-mdx

Comment (AhmedMuhammedElsaid) [reply examples/with-mdx/package.json:10]: True, it's being covered there.

Comment (remcohaszing) [reply examples/with-mdx/package.json:10]: The benefit that `@mdx-js/react` offers, is that it allows users to provide components for MDX files for different files in the React tree. IMO this is a niche use case. I think the `mdx-components` file is a much better solution for Next.js. Kudos to whoever came up with that.

`@mdx-js/react` works for the pages directory, but not for the app directory. It just crashes with a weird error. I personally found this confusing and I’ve heard so from others too.

Regardless of what works or doesn’t work, IMO it’s good to have a single blessed way to use MDX with Next.js. I think it’s hard to explain what the type of router has to do with this.

Using the `mdx-components` file better aligns with the DX after I implement [support for provided components](https://github.com/mdx-js/mdx-analyzer/issues/399) in the MDX language server. Just define them once globally, then use them in every MDX file.

What is missing from this PR though, is the addition of the `mdx-components` file. I could add it, but even better would be to merge #59693.

Comment (leerob) [reply examples/with-mdx/package.json:10]: FYI: The `mdx-components` file is only for the App Router, not for the Pages Router.

Comment (leerob) [reply examples/with-mdx/package.json:10]: Interesting... good point! TIL

**Comment by karlhorky** [comment]
@samcx would it be possible to get another review of this PR?

**Review by samcx — APPROVED** [review]
Thank you for submitting a PR!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Pull request merged**
